app/scripts/ui.js

```diff
@@ -11,19 +11,62 @@ import Eth from 'ethjs';
 import EthQuery from 'eth-query';
 import StreamProvider from 'web3-stream-provider';
 import log from 'loglevel';
+import fetch from 'node-fetch';
 import launchMetaMaskUi from '../../ui';
 import {
   ENVIRONMENT_TYPE_FULLSCREEN,
   ENVIRONMENT_TYPE_POPUP,
 } from '../../shared/constants/app';
+import { SUPPORT_LINK } from '../../ui/helpers/constants/common';
 import ExtensionPlatform from './platforms/extension';
 import { setupMultiplex } from './lib/stream-utils';
 import { getEnvironmentType } from './lib/util';
 import metaRPCClientFactory from './lib/metaRPCClientFactory';
+import getFirstPreferredLangCode from './lib/get-first-preferred-lang-code';
 
 start().catch(log.error);
 
 async function start() {
+  const preferredLocale = await getFirstPreferredLangCode();
+
+  const url = browser.runtime.getURL(
+    `../_locales/${preferredLocale}/messages.json`,
+  );
+  const messages = await (await fetch(url)).json();
+
+  function displayCriticalError(container, err, msg) {
+    const html = `
+    <style>
+      #app-content {margin: 10px;}
+      .critical-error {
+        color:red; 
+        background-color:yellow; 
+        text-align: left
+      }
+      blockquote {
+        margin: 20px 0 20px 5px; 
+        padding-left: 5px; 
+        border-left: 1px solid grey
+      }  
+      a {color: blue}
+      a:hover {color: green}
```

Comment:
Could we borrow CSS styles from the `ActionableMessage` component so this error looks uniform to other errors we present to the user?

Reply:
I don't think so, because I don't think we can use React here.

Reply:
The CSS isn't added by React. They're unrelated. The CSS is bundled into a single file and added to the document separately from the JS, in the `<head>` actually so it should load first I think.

Reply:
It does break conventions to use a component classname outside of that component though. Maybe we can lift or copy the error styles into a global class, something in `ui/css`

Reply:
good point, so disregard my comment. Good suggestions here from Mark